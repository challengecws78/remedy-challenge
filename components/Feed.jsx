import React, { Component } from 'react';
import axios from 'axios';
import Link from 'next/link';

export class Feed extends Component {
    constructor(props) {
        super(props)
        this.state = {
            guests: []
        }
    }

    async componentDidMount() {
        try {
            const res = await axios.get('/api/guestbook');
            this.setState({
                guests: res.data.posts
            })
        } catch(e) {
            console.log(e);
        }
        
    }

    get guestbookList() {
        return this.state.guests.map(guest => (
            <div>
                <h3>{guest.name}</h3>
                <p>{guest.message}</p>
            </div>
        ));
    }

    render() {
        return (
            <div>
            <Link href="/">Guest Form</Link>
            {this.guestbookList}
            </div>
        )
    }
    
}

export default Feed;