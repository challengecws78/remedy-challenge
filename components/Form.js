import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ValidatedInput from '../Inputs/ValidatedInput';

export class ContactForm extends Component {
    constructor(props) {
        super(props)
        this.state = {
            name: '',
            message: '',
        }
    }
    

    static propTypes = {
        onSubmit: PropTypes.func.isRequired,
        name: PropTypes.string,
        message: PropTypes.string,
    }
    

    handleChange = ({ name, value }) => {
        this.setState({
            [name]: value
        });
    }

    get getValidation () {
        const { name, message } = this.state;
        const nameLength = name.length;
        const messageLength = message.length;


        if (nameLength > 0 && messageLength > 0) {
            return false;
        } else {
            return true;
        }
    }

    onSubmit = e => {
        e.preventDefault();
        const { name, message } = this.state;
        this.props.onSubmit({
            name,
            message,
        });
    }
    
    render() {
        return (
            <div>
                <form className="contact-form" onSubmit={this.onSubmit}>
                    <ValidatedInput
                        label="Name" 
                        name="name"
                        type="text"
                        placeholder="Name"
                        value={this.state.firstName}
                        onChange={this.handleChange}
                        textArea={false}
                        validate={val => (val ? false : 'Name is Required')}
                    />
                    <ValidatedInput
                        label="Message" 
                        name="message"
                        type="text"
                        placeholder="Message"
                        value={this.state.lastName}
                        onChange={this.handleChange}
                        textArea={false}
                        validate={val => (val ? false : 'Message is Required')}
                    />
                    <button type="submit" className="btn btn-primary" disabled={this.getValidation}>{this.props.title}</button>
                </form>
            </div>
        );
    }
}



export default ContactForm;