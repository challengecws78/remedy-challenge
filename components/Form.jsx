import React, { Component } from 'react';
import ValidatedInput from './Inputs/ValidatedInput';
import Link from 'next/link';
import axios from 'axios';

export class Form extends Component {
    constructor(props) {
        super(props)
        this.state = {
            name: '',
            message: '',
            status: ''
        }
    }

    handleChange = ({ name, value }) => {
        this.setState({
            [name]: value,
            status: ''
        });
    }

    get getValidation () {
        const { name, message } = this.state;
        const nameLength = name.length;
        const messageLength = message.length;


        if (nameLength > 0 && messageLength > 0) {
            return false;
        } else {
            return true;
        }
    }

    handleSubmit = e => {
        e.preventDefault();
        const { name, message } = this.state;
        axios.post('/api/guestbook', { name, message });
        this.setState({
            name: '',
            message: '',
            status: 'Success!'
        });
    }
    
    render() {
        return (
            <div>
                <Link href="/guestbook">Feed</Link>
                <form className="contact-form" onSubmit={this.handleSubmit}>
                    <ValidatedInput
                        label="Name" 
                        name="name"
                        type="text"
                        placeholder="Name"
                        value={this.state.name}
                        onChange={this.handleChange}
                        textArea={false}
                        validate={val => (val ? false : 'Name is Required')}
                    />
                    <ValidatedInput
                        label="Message" 
                        name="message"
                        type="text"
                        placeholder="Message"
                        value={this.state.message}
                        onChange={this.handleChange}
                        textArea={true}
                        validate={val => (val ? false : 'Message is Required')}
                    />
                    <button type="submit" className="btn btn-primary" disabled={this.getValidation}>Add to Guest Book</button>
                    <span style={{ color: 'green' }}>{this.state.status}</span> 
                </form>
            </div>
        );
    }
}



export default Form;