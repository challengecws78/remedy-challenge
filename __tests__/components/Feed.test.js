import React from 'react';
import { shallow, mount } from 'enzyme';
import { Feed } from '../../components/Feed';

jest.mock('axios', () => {
    const exampleGuestBook = { posts: [
      { name: 'Aaron Rodgers', message: 'cheese' }
    ]};
    
    return {
      get: jest.fn(() => Promise.resolve(exampleGuestBook)),
    };
});

const axios = require('axios');

describe('<FeedList />', () => {
    it('should render FeedList with empty message', () => {
        const wrapper = shallow(<Feed />);
        expect(wrapper).toMatchSnapshot();
    });

    it('fetch guests on componentDidMount', async (done) => {
        const wrapper = await mount(<Feed />);
          wrapper
          .instance()
          .componentDidMount()
            expect(axios.get).toHaveBeenCalled();
            expect(axios.get).toHaveBeenCalledWith('/api/guestbook');
            done()
            wrapper.update();
 
           expect(wrapper.state('guests')).toEqual([{ name: 'Aaron Rodgers', message: 'cheese' }]);
      });
})