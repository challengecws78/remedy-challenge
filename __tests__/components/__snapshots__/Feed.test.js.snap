// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedList /> should render FeedList with empty message 1`] = `
<div>
  <Link
    href="/"
  >
    Guest Form
  </Link>
</div>
`;
